spec/lib/carma/client/client_spec.rb

```diff
@@ -4,6 +4,19 @@
 require 'carma/client/client'
 
 RSpec.describe CARMA::Client::Client, type: :model do
+  let(:restforce_client) do
+    restforce_client = double
+    expect(subject).to receive(:get_client).and_return(restforce_client)
+
+    if Settings['salesforce-carma'].mock
```

Comment:
I think tucking away this conditional statement in this variable definition, that is used for multiple tests, makes it harder to understand what's happening, and when. It also takes away from the ability to read down the page linearly to understand what mocks cause which expectations.

Since these expectations happen **as a result** of `Settings['salesforce-carma'].mock`, I'd keep them close to where that variable is being mocked. Keeping related context close together makes it much easier to read linearly and understand that there is a causal relationship between that mock and these expectations.

```rb
context 'with betamocks enabled' do
  before do
    expect(Settings['salesforce-carma']).to receive(:mock).and_return(true)

    expect(restforce_client).to receive(:builder).and_return(builder)
    expect(builder).to receive(:insert_before).with(Faraday::Adapter::NetHttp, Betamocks::Middleware)
  end

  it "does the thing"
    # ... (test with it enabled)
  end
end

context 'with betamocks disabled' do
  before do
    expect(Settings['salesforce-carma']).to receive(:mock).and_return(false)
    expect(restforce_client).not_to receive(:builder)
  end

  it "does the thing"
    # ... (test with it disabled)
  end
end
```